Wikipedia/Code/HistoryViewController.swift

```diff
@@ -1,306 +1,377 @@
-        let profileButtonConfig: WMFNavigationBarProfileButtonConfig?
-        let tabsButtonConfig: WMFNavigationBarTabsButtonConfig?
+
         if let dataStore {
             profileButtonConfig = self.profileButtonConfig(target: self, action: #selector(userDidTapProfile), dataStore: dataStore, yirDataController: yirDataController, leadingBarButtonItem: nil)
             tabsButtonConfig = self.tabsButtonConfig(target: self, action: #selector(userDidTapTabs), dataStore: dataStore, leadingBarButtonItem: deleteButton)
-        } else {
-            profileButtonConfig = nil
-            tabsButtonConfig = nil
         }
 
         configureNavigationBar(titleConfig: titleConfig, closeButtonConfig: nil, profileButtonConfig: profileButtonConfig, tabsButtonConfig: tabsButtonConfig, searchBarConfig: nil, hideNavigationBarOnScroll: hideNavigationBarOnScroll)
-    }
-    
-    private func updateProfileButton() {
 
-        guard let dataStore else {
-            return
-        }
-
-        let config = self.profileButtonConfig(target: self, action: #selector(userDidTapProfile), dataStore: dataStore, yirDataController: yirDataController, leadingBarButtonItem: nil)
-        updateNavigationBarProfileButton(needsBadge: config.needsBadge, needsBadgeLabel: CommonStrings.profileButtonBadgeTitle, noBadgeLabel: CommonStrings.profileButtonTitle)
     }
 
     @objc func userDidTapProfile() {
-        
         guard let dataStore else {
             return
         }
-        
+
         guard let languageCode = dataStore.languageLinkController.appLanguage?.languageCode,
               let metricsID = DonateCoordinator.metricsID(for: .historyProfile, languageCode: languageCode) else {
             return
         }
-        
+
         DonateFunnel.shared.logHistoryProfile(metricsID: metricsID)
-        
+
         profileCoordinator?.start()
     }
-    
+
     @objc func userDidTapTabs() {
-        _ = tabsCoordinator?.start()
+        tabsCoordinator?.start()
         ArticleTabsFunnel.shared.logIconClick(interface: .history, project: nil)
     }
 
-    func titleForHeaderInSection(_ section: Int) -> String? {
-        guard let sections = fetchedResultsController.sections, sections.count > section else {
-            return nil
+    func tappedArticle(_ item: HistoryItem) {
+        if let articleURL = item.url, let dataStore, let navVC = navigationController {
+            let articleCoordinator = ArticleCoordinator(navigationController: navVC, articleURL: articleURL, dataStore: dataStore, theme: theme, source: .history)
+            articleCoordinator.start()
         }
-        let sectionInfo = sections[section]
-        guard let article = sectionInfo.objects?.first as? WMFArticle, let date = article.viewedDateWithoutTime else {
-            return nil
+    }
+
+    func share(item: HistoryItem, frame: CGRect?) {
+        if let dataStore, let url = item.url {
+            let article = dataStore.fetchArticle(with: url)
+            if let frame {
```

Comment:
I'm not sure where the bug might be, but my share frames aren't lining up on my iPad mini:

![Search 2](https://github.com/user-attachments/assets/5f22d333-c3ca-498f-8965-e4bee6031c27)